2333/2/libs/analyzer.rb

```diff
@@ -0,0 +1,47 @@
+require_relative 'handler'
+require 'terminal-table'
+require_relative './analyzer_printer.rb'
+require_relative './names_searcher.rb'
+
+# VersusBattle text analyzer
+class Analyzer
+  def initialize(rappers, take_value = nil, selected_name = nil)
+    @rappers = rappers
+    @take_value = take_value.to_i
+    @name = selected_name
+  end
+
+  def top_bad_words
+    AnalyzerPrinter.print_top_bad_words(analyze_rappers)
```

Comment:
The method `#analyze_rappers` looks good, but the bad part of it is that it calls other methods with arguments:
```rb
def analyze_rappers
  # ...
  sort_by_mats(rappers)
end

def sort_by_mats(rappers)
  # ...
end
```

So you kind of pass the same variable further and further. 

We can do better ;-).

Instead of passing variables to a methods, name them appropriately and work with instance variables or other methods:

```rb
def table_rows_of_sorted_rappers # consider moving this to another class ;).
  top_n_rappers_sorted_by_bad_words.map # ...
end

def top_n_rappers_sorted_by_bad_words
  # ...
  @rappers.sort_by # ... .limit(@take_value) 
end
```